### PR TITLE
live tab rendering issue

### PR DESCRIPTION
**Pull request by mehaknasir**
live tab configured alone was rendering unusual UI behavior and that issues is fixed in this PR

https://user-images.githubusercontent.com/67791278/154243868-65d56586-06e7-4dcc-9c0f-893cfa6e95ed.mov

### CODE REVIEW

File: src/components/program/ProgramPage.jsx [hunk 1]
```diff
@@ -49,12 +49,15 @@ class ProgramPage extends Component {
...
       this.validateProgramAccess(enrolledPrograms);
     }
-    if (tabViewEnabled && tabViewEnabled !== prevProps.tabViewEnabled) {
-      // eslint-disable-next-line max-len
-      const switchTabView = tabViewEnabled && (programDiscussions.configured || liveSettings.configured);
+    if ((tabViewEnabled && tabViewEnabled !== prevProps.tabViewEnabled)
+      || programDiscussions.configured !== prevProps.programDiscussions.configured
+      || liveSettings.configured !== prevProps.liveSettings.configured) {
+      const switchTabView = tabViewEnabled
+        && (programDiscussions.configured || liveSettings.configured);
```
Comment (awaisdar001) [inline src/components/program/ProgramPage.jsx:60]: Didn't the API return the data that tab view is enabled for this program? why you are checking the logic here?

Comment (awaisdar001) [inline src/components/program/ProgramPage.jsx:60]: naming can be improved here too. `liveSettings` => `programLiveTab` and `programDiscussions` => `programDiscussionTab`

**Review by awaisdar001 — COMMENTED** [review]
Added a few questions here.

File: src/components/program/ProgramPage.jsx (hunk 1 above)
Comment (mehaknasir) [reply src/components/program/ProgramPage.jsx:60]: there are multiple apis each returning its own part of data what is configured and whats not. frontend needs to check commulatively if we need to switch to tab view or not so lets suppose

```
flag : enabled
live: disabled
discussion: disbaled
```

so although flag is enabled but we cant switch to tab view.

[the remark this answers is not in the document]
Comment (mehaknasir) [reply src/components/program/ProgramPage.jsx:60]: we have access to this `prevProps` in the only componentDidUpdate the lifecycle hook if we pass this if logic in a function we need to pass that argument I don't feel that's the right way.

Comment (mehaknasir) [reply src/components/program/ProgramPage.jsx:60]: let me check what impact it has on renaming at this stage, as the variables are used in multiple places and bound with redux as well.

**Pull request merged**